### PR TITLE
[All devs approval required] Update react-redux to the latest 7.1 with hooks support

### PR DESCRIPTION
**Pull request by desfero**
**Technical notes**
This one updates react-redux (and some related dependencies) to the latest versions. The most notable changes are hooks api for redux. They also claim that react-redux should be also a little bit faster after the update.

**Notable API changes**
Hooks api for redux - https://react-redux.js.org/api/hooks
batch method for dispatching actions - https://react-redux.js.org/api/batch

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -18082,24 +18073,24 @@ react@16.9.0:
     object-assign "^4.1.1"
     prop-types "^15.6.2"
 
-react@^16.2.0, react@^16.8.6:
-  version "16.8.6"
-  resolved "https://registry.yarnpkg.com/react/-/react-16.8.6.tgz#ad6c3a9614fd3a4e9ef51117f54d888da01f2bbe"
-  integrity sha512-pC0uMkhLaHm11ZSJULfOBqV4tIZkx87ZLvbbQYunNixAAvjnC+snJCg0XQXn9VIsttVsbZP/H/ewzgsd5fxKXw==
+react@^16.8.3:
```
Comment (Moejoe90) [inline yarn.lock]: I wonder why we have to different react versions

File: packages/web/app/utils/withActionWatcher.tsx [hunk 2]
```diff
@@ -25,8 +25,8 @@ interface IWatchActionOptions {
 export const withActionWatcher: (
   options: IWatchActionOptions,
 ) => (
-  WrappedComponent: React.ComponentType,
-) => React.ComponentClass = options => WrappedComponent =>
+  WrappedComponent: React.ComponentType<unknown>,
```
Comment (Moejoe90) [inline packages/web/app/utils/withActionWatcher.tsx]: :1st_place_medal:

File: yarn.lock (hunk 1 above)
Comment (arjunu) [reply yarn.lock]: there are actually three

**Comment by desfero** [comment]
Closing for now until https://github.com/supasate/connected-react-router/issues/311 get's fixed

**Pull request closed without merging**